[mcp23017] Binding is not working on OH2.5

After installing the addon through PaperUI, the following error appears. If someone cold resolve this, it would be nice. The addon right now does not work.

```
2019-09-18 22:28:06.573 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.dispose()' on 'org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler@bdf7ba': null
java.lang.NullPointerException: null
	at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.dispose(Mcp23017Handler.java:186) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [133:org.openhab.core:2.5.0.M3]
	at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [133:org.openhab.core:2.5.0.M3]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
2019-09-18 22:28:06.579 [ERROR] [core.thing.internal.ThingManagerImpl] - Exception occurred while disposing handler of thing 'mcp23017:mcp23017:27599645': null
java.lang.NullPointerException: null
	at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.dispose(Mcp23017Handler.java:186) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:?]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [133:org.openhab.core:2.5.0.M3]
	at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:53) [133:org.openhab.core:2.5.0.M3]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?] 
```


* Version used: Openhab 2.5-M3 installed by Openhabian
* Operating System RPi 3

So I took the Snapshot from here (https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/addons/bundles/org.openhab.binding.mcp23017/2.5.2-SNAPSHOT/), placed in the folder I mentioned above and it's not working.

``

2020-01-29 17:47:08.115 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler@1aa8eff': null

java.lang.NullPointerException: null

	at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.handleOutputCommand(Mcp23017Handler.java:116) ~[?:?]

	at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.handleCommand(Mcp23017Handler.java:83) ~[?:?]

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_222]

	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_222]

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_222]

	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_222]

	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]

	at org.eclipse.smarthome.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59) [bundleFile:?]

	at com.sun.proxy.$Proxy190.handleCommand(Unknown Source) [?:?]

	at org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:74) [bundleFile:?]

	at org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:48) [bundleFile:?]

	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_222]

	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_222]

	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_222]

	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_222]

	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]

	at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]

	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_222]

	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_222]

	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_222]

	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
``

Hi,
I updated from OH 2.4 to OH 2.5 and I have the problem in calling handleCommand() method since then:
`2020-02-09 08:07:39.485 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler@cee989': null
java.lang.NullPointerException: null
	at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.handleOutputCommand(Mcp23017Handler.java:116) ~[?:?]
	at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.handleCommand(Mcp23017Handler.java:83) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_152]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_152]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
	at org.eclipse.smarthome.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59) [bundleFile:?]
	at com.sun.proxy.$Proxy446.handleCommand(Unknown Source) [?:?]
	at org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:74) [bundleFile:?]
	at org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:48) [bundleFile:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_152]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_152]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
	at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
	at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_152]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_152]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_152]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]`

What further information/data would you need to re-open the ticket? I believe it is a generic error, irrelevant of user data and comparing the 2.4.0 vs. 2.5.x would be the way to go. Thank you.

from configuration files: 
relevant items: 
```
Switch          F2_MasterBedroom_Window_Down "Window Down"      <rollershutter>   (gShutter)      {channel="mcp23017:mcp23017:chip20:output#B4"}
Switch          F2_MasterBedroom_Window_Up  "Window Up"        <rollershutter>   (gShutter)      {channel="mcp23017:mcp23017:chip20:output#B6"}
```

relevant things:
```
Thing mcp23017:mcp23017:chip20  "mcp1" [address=20,bus=1] {
    Type output_pin : output#A0 [default_state="HIGH"]
    Type output_pin : output#A1 [default_state="HIGH"]
    
    Type output_pin : output#B0 [default_state="LOW"]

    Type output_pin : output#B3 [default_state="LOW"]
    Type output_pin : output#B4 [default_state="LOW"]
    Type output_pin : output#B5 [default_state="LOW"]
    Type output_pin : output#B6 [default_state="LOW"]
}
```

rule sending command: 
```
rule "Bedroom Window Down"
when
  Item F2_MasterBedroom_Window received command DOWN
then
  logInfo("blind_bedroom.rules", "Bedroom window Blind down")
  F2_MasterBedroom_Window_Up.sendCommand(OFF) // motor protection
  F2_MasterBedroom_Window_Down.sendCommand(ON)
  var howLongToRun = 20;
  if (winterMode.state == OFF) {
    howLongToRun = 15;
  }
  createTimer(now.plusSeconds(howLongToRun), [ | F2_MasterBedroom_Window_Down.sendCommand(OFF) ])
end
```

ouput of log:tail from karaf, after changing to DEBUG for org.openhab:

```
19:10:39.800 [DEBUG] [3017.internal.handler.Mcp23017Handler] - Received command: ON on channelGroup output on channel B4
19:10:39.810 [DEBUG] [3017.internal.handler.Mcp23017Handler] - Defined channel in group: output. Valid: true
19:10:39.821 [DEBUG] [3017.internal.handler.Mcp23017Handler] - Is channel B4 in supported channels: true
19:10:39.832 [DEBUG] [ternal.handler.Mcp23017PinStateHolder] - Getting output pin for channel mcp23017:mcp23017:chip20:output#B4
19:10:39.842 [DEBUG] [ternal.handler.Mcp23017PinStateHolder] - initializeOutputPin for channel mcp23017:mcp23017:chip20:output#B4
19:10:39.852 [DEBUG] [ternal.handler.Mcp23017PinStateHolder] - initializing for pinState LOW
19:10:39.869 [DEBUG] [ternal.handler.Mcp23017PinStateHolder] - Bound digital output for PIN: GPIO B4, channel: mcp23017:mcp23017:chip20:output#B4, pinState: LOW
19:10:39.881 [ERROR] [rnal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.handleCommand()' on 'org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler@5ff410': null
java.lang.NullPointerException: null
        at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.handleOutputCommand(Mcp23017Handler.java:116) ~[?:?]
        at org.openhab.binding.mcp23017.internal.handler.Mcp23017Handler.handleCommand(Mcp23017Handler.java:83) ~[?:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_152]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_152]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
        at org.eclipse.smarthome.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59) [bundleFile:?]
        at com.sun.proxy.$Proxy187.handleCommand(Unknown Source) [?:?]
        at org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:74) [bundleFile:?]
        at org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:48) [bundleFile:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_152]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_152]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_152]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_152]
        at org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152) [bundleFile:?]
        at org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_152]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_152]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_152]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]
19:10:40.544 [DEBUG] [hab.binding.http.internal.HttpBinding] - 'execute()' called ...

```

In the meanwhile I also looked into the code and if the problem happens in line 116 (providing the line number didn't change from 2.5.1 to 2.5.2-SNAPSHOT (actual code I forked)), then this is the line:
`boolean activeLowFlag = ACTIVE_LOW_ENABLED.equalsIgnoreCase(configuration.get(ACTIVE_LOW).toString());`

I looked into my configuration (REST API) and I don't see 'active_low' in my channels. 
So either the _configuration_ is null or the _configuration.get(ACTIVE_LOW)_ delivers null, am I right?

--- a/bundles/org.openhab.binding.mcp23017/src/main/java/org/openhab/binding/mcp23017/internal/handler/Mcp23017Handler.java
+++ b/bundles/org.openhab.binding.mcp23017/src/main/java/org/openhab/binding/mcp23017/internal/handler/Mcp23017Handler.java
@@ -15,6 +15,7 @@ package org.openhab.binding.mcp23017.internal.handler;
 import static org.openhab.binding.mcp23017.internal.Mcp23017BindingConstants.*;
 
 import java.io.IOException;
+import java.util.Objects;
 
 import org.eclipse.smarthome.config.core.Configuration;
 import org.eclipse.smarthome.core.library.types.OnOffType;
@@ -113,7 +114,8 @@ public class Mcp23017Handler extends BaseThingHandler implements GpioPinListener
             Configuration configuration = this.getThing().getChannel(channelUID.getId()).getConfiguration();
 
             // invertLogic is null if not configured
-            boolean activeLowFlag = ACTIVE_LOW_ENABLED.equalsIgnoreCase(configuration.get(ACTIVE_LOW).toString());
+            String activeLowStr = Objects.toString(configuration.get(ACTIVE_LOW), null);
+            boolean activeLowFlag = ACTIVE_LOW_ENABLED.equalsIgnoreCase(activeLowStr);
             PinState pinState = command == OnOffType.ON ^ activeLowFlag ? PinState.HIGH : PinState.LOW;
             logger.debug("got output pin {} for channel {} and command {} [active_low={}, new_state={}]", outputPin,
                     channelUID, command, activeLowFlag, pinState);

